travharv/web_discovery.py

```diff
@@ -89,37 +91,46 @@ def get_description_into_graph(
     session.mount("http://", adapter)
     session.mount("https://", adapter)
 
-    # TODO actually use the format arg to select types and load them _all_
-    #      (not just the server pick as below) see --> issue #31
-    headers = {"Accept": "text/turtle ,application/ld+json"}
-    r = session.get(subject_url, headers=headers)
-
-    # check if the request was successful and it returned a json-ld or ttl file
-    if r.status_code == 200 and (
-        "application/ld+json" in r.headers["Content-Type"]
-        or "text/turtle" in r.headers["Content-Type"]
-        or "application/json" in r.headers["Content-Type"]
-    ):
-        # parse the content directly into the triplestore
-        if "text/turtle" in r.headers["Content-Type"]:
-            format = "turtle"
-        elif (
-            "application/ld+json" in r.headers["Content-Type"]
-            or "application/json" in r.headers["Content-Type"]
-        ):
-            format = "json-ld"
+    triples_found = False
 
-        try:
-            graph.parse(data=r.text, format=format, publicID=subject_url)
-            log.info(
-                f"content of {subject_url} added to triplestore in {format=}"
-            )
-        except Exception as e:
-            log.warning(
-                f"failed to parse {subject_url} in {format=} error: {e}"
-            )
+    ACCEPTABLE_MIMETYPES = {
+        "application/ld+json",
+        "text/turtle",
+        "application/json",
+        "application/octet-stream",  # This temp for testing
```

Comment:
uuuurgh? feels odd
code related to tests should only be in tests/**py

and surely does not reed as an ACCEPTABLE (rdf-containing?) mime-type